### ISSUE TITLE
canceling an enqueued job

### ISSUE THREAD

**Issue by guyisra** [utgarda/sidekiq-status#50]
From the code, I understand that only scheduled jobs can be unscheduled..

Is there a way to cancel/delete enqueued jobs that are yet to be push to workers?

**Comment by utgarda** [comment]
All the jobs you put to Redis through your client are "scheduled" - see that in Sidekiq client code: https://github.com/mperham/sidekiq/blob/master/lib/sidekiq/client.rb#L177 , but those not scheduled for the future, are scheduled to be executed ASAP, which means they get executed right after all the "right now" jobs issued before them.

But why do you want that? Maybe there's an easier way to do what you need.

**Comment by guyisra** [comment]
ok, so in this specific use case a user instruct a controller to process images in a certain way.
This results in a single worker and then, based on some logic, would create more workers to be executed (it can be few workers or hundreds/thousands depending on the user's instruction)

on the client, the users sees the resulting images every few seconds, and sometimes it is needed to stop this execution and switch to a different set of images (which will also results in a lot of workers, immediately scheduling).

In those workers, I first check whether the user decided to stop, and if so exit immediately, so the jobs run fairly quickly, but they do run. I want to save any job that relates to a particular image set, so I can, when the user decided to stop processing, cancel/dequeue all those jobs so they won't even start running. 

Sidekiq can do this inefficiently if I go through the queue and find every job id. 

So, basically, I have a bunch of specific jobs waiting to be moved to workers and I want them gone, so they won't start running. I can't just clear the queue since that will delete jobs of other users / image sets.

Is it possible?

**Comment by utgarda** [comment]
Oops, I mesread sidekiq's code, not everything is "scheduled', most of the jobs go just to their respective queues: https://github.com/mperham/sidekiq/blob/master/lib/sidekiq/client.rb#L196

How about creating separate queues for every user? Yeah, the docs don't recommend that https://github.com/mperham/sidekiq/wiki/Advanced-Options#queues but why don't you check how bad is the penalty for that in your case?
